### DISCUSSION TITLE
Authentication

### DISCUSSION THREAD

**Discussion by yesoreyeram** [grafana/grafana-infinity-datasource#40] [General]
# Refer https://yesoreyeram.github.io/grafana-infinity-datasource/wiki/authentication for the latest guidelines

**Comment by Orch3strator** [comment]
Hello,

I have to login via REST API and get a bearer token, which I have to us in the subsequent query to authenticate. How can I utilize this feature? Is this part of "Release v0.7.0-alpha0"?

Great Project!

**Reply by yesoreyeram** [reply]
Yes. The above feature is available in 0.7 alpha and looking for feedback. Docs are not available though.

**Reply by Orch3strator** [reply]
Thank you,

I was not able to get it to show up.
Can you provide some screenshots and steps?

Regards, V.

**Reply by Orch3strator** [reply]
Here are the steps I have to take:

- post credentials to /login url 
- the result is the authentication token
- use this token for all other REST API calls to the same server. I don't want to have to login for each API call if possible.

Regards, V.

**Reply by yesoreyeram** [reply]
Hi @Orch3strator - sorry to hear that you are having trouble and accept my apologies for the delayed response. This feature existed only in the 0.7-alpha0 version and then discontinued due to design changes.

**Reply by ngui1975** [reply]
Hi Sriram yesoreyeram,
Thanks for your plugin is very useful, but I have same problem. There is any way to query authentication API and reset Bearer Token every 24 hours?

Best regards,
Nelson

**Comment by ddramireddy** [comment]
Hi @yesoreyeram ,

Is this feature is supported in any of the releases? I am planning to use this plugin for our API, which uses bearer token. If this is not released, I would like to use this branch and install this plugin. Is there any developer documentation, I can refer to generate executable out of this source code.

**Reply by yesoreyeram** [reply]
Hi @ddramireddy - this feature is now deferred for development. Also, current main is branched out in different direction. So the upcoming token authentication design will be something different when it comes live.

But if this branch is useful, feel free to use it and enhance it as per your need. And extremelly sorry as there is no documentation as I am the only one contributor at the moment. But this is like any other grafana's datasource plugin. this is straight forward like `yarn install` and `yarn watch` for development purposes.

**Reply by ddramireddy** [reply]
Hi Sriram,

Thank you for prompt response. Able to compile and load this plugin with following instructions.


```
yarn install --pure-lockfile  

yarn build

mv dist/ yesoreyeram-infinity-datasource 

zip yesoreyeram-infinity-datasource.zip yesoreyeram-infinity-datasource -r

 grafana-cli plugins ls 

cd /usr/local/var/lib/grafana/plugins

cp ~/Downloads/grafana-infinity-datasource-feature-token-authentication/yesoreyeram-infinity-datasource.zip /usr/local/var/lib/grafana/plugins/

unzip yesoreyeram-infinity-datasource.zip

 brew services restart grafana

tail -f  /usr/local/var/log/grafana/grafana.log 

```

**Reply by ddramireddy** [reply]
Hi @yesoreyeram ,

I am able to configure this branch and started using TokenAuth successfully. One issue I observed is of variables. While creating variables, I do see only "Legacy" option in Query type dropdown. But, the following link says, "Infinity" query type also is supported.

https://sriramajeyam.com/grafana-infinity-datasource/wiki/template-variables
 
But I don't see this option. Is it the case of this TokenAuth feature branch is not in sync with main branch? Is it possible to get latest and update this feature branch? Not sure, whether I have permissions to do this.

**Reply by yesoreyeram** [reply]
You are right. your branch doesn't have `Infinity` where the main or 0.7.x has. You dont need permissions to do that. You can merge on your own fork.

**Reply by solly4by7** [reply]
Hi is there any doc on how we can configure for REST API's with OAuth2?

**Reply by yesoreyeram** [reply]
No. Oauth is not supported as of now

**Reply by solly4by7** [reply]
Do you know any other plugins that can help me pull data through REST API's with OAuth2 to Grafana.

**Reply by yesoreyeram** [reply]
No I am not aware of. Are you interested in any particular api? I am looking for sample APIs to build a generic oauth experience. If you can provide sample applications that use oauth, that will be helpful to build this experience
